cardano-api/src/Cardano/Api/Typed.hs

```diff
@@ -2211,6 +2250,110 @@ deserialiseFromRawBytesHex proxy hex =
       | otherwise        -> Nothing
 
 
+-- ----------------------------------------------------------------------------
+-- Bech32 Serialisation
+--
+
+class (HasTypeProxy a, SerialiseAsRawBytes a) => SerialiseAsBech32 a where
+  humanReadablePrefix :: a -> Bech32.HumanReadablePart
+
+  serialiseToBech32 :: a -> Either Bech32EncodeError Text
+  serialiseToBech32 a =
+    first Bech32EncodingError
+      . Bech32.encode (humanReadablePrefix a)
```

Comment:
We'll need to use the lenient variant, `Bech32.encodeLenient`, as it appears that addresses containing both a payment and stake credential end up being longer than 90 characters once encoded.